src/main/java/io/jenkins/plugins/audit/listeners/UserLogListener.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package io.jenkins.plugins.audit.listeners;
+
+import hudson.ExtensionList;
+import hudson.Extension;
+import jenkins.security.SecurityListener;
+import io.jenkins.plugins.audit.RequestContext;
+
+import javax.annotation.Nonnull;
+import java.net.InetAddress;
+import java.time.Instant;
+
+import io.jenkins.plugins.audit.event.Login;
+import io.jenkins.plugins.audit.event.Logout;
+import org.apache.logging.log4j.audit.LogEventFactory;
+import org.apache.logging.log4j.core.util.NetUtils;
+
+/**
+ * Listener notified of user login and logout events.
+ */
+ @Extension
+public class UserLogListener extends SecurityListener {
+
+    Login login = LogEventFactory.getEvent(Login.class);
```

Comment:
I don't think any of these fields should be fields. Shared instances of data classes across threads is not a good idea without synchronization, and I'm not so sure if we really need synchronization here if we just create these values inside the methods.

Reply:
i'll put them inside the methods then.